File: src/features/typing/data/go/maximum_subarray.go
package main

func maxSubArray(nums []int) int {
	best := nums[0]
	sum := 0
	for _, num := range nums {
		if sum < 0 {
			sum = 0
		}
		sum += num
		if sum > best {
			best = sum
		}
	}
	return best
}

File: src/features/typing/data/ts/maximum_subarray.ts
function maxSubArray(nums: number[]): number {
  let best = nums[0];
  let sum = 0;
  for (const num of nums) {
    if (sum < 0) {
      sum = 0;
    }
    sum += num;
    best = Math.max(best, sum);
  }
  return best;
}
